### PR TITLE
Marom Goldshmidt PR

### PR DESCRIPTION
**Pull request by MaromGold**
This is last week's task...

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
+# The docker part could also be accomplished using: sudo apt install docker && sudo systemctl start docker && sudo systemctl enable docker
+curl -fsSL https://get.docker.com -o get-docker.sh
+sh get-docker.sh
+
+cd ~
```
Comment (greycat001) [inline bootstrap.sh]: Using '~' in scripts is bad practice. It may yield unpredictable results.

File: nginx.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	fi
+
+else
+    echo "nginx does not exist on this machine"
+    echo "Installing nginx now..."
+    sudo yum install -y nginx
+    sudo systemctl start nginx
+    sudo systemctl enable nginx
+fi
+#test
```
Comment (greycat001) [inline nginx.sh]: What is this line for?

File: ncdu.sh [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    $ncdu_status == "1"
+    echo -e "Install? (y/n) \c"
+
+if "$REPLY" = "y"; then
+
+      sudo yum install -y ncdu
+      sudo systemctl start ncdu
+      sudo systemctl enable ncdu
+   fi
+fi
```
Comment (greycat001) [inline ncdu.sh]: No newline at end of file.
This may lead to unpredictable error messages after the script finishes.

File: bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,18 @@
+#!/bin/sh
```
Comment (greycat001) [inline bootstrap.sh]: Why not bash as in other scripts?

**Review by greycat001 — CHANGES_REQUESTED** [review]
The scripts look nice.
All of them lack a short description of their purpose at the beginning.

File: bootstrap.sh [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+sh get-docker.sh
+
+cd ~
+sudo wget -O /etc/yum.repos.d/jenkins.repo https://pkg.jenkins.io/redhat-stable/jenkins.repo
+sudo rpm --import https://pkg.jenkins.io/redhat-stable/jenkins.io.key
+yum install jenkins
+sudo systemctl start jenkins.service
+sudo systemctl enable jenkins.service
+
+wget https://github.com/git/git/archive/v2.29.1.tar.gz -O git.tar.gz
```
Comment (lmilbaum) [inline bootstrap.sh]: Was git installation required in the first home challenge?

File: nginx.sh [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+	fi
+    if "$choice" = "no"; then echo "OK. nginx will remain intact".
+	fi
+
+else
+    echo "nginx does not exist on this machine"
+    echo "Installing nginx now..."
+    sudo yum install -y nginx
+    sudo systemctl start nginx
+    sudo systemctl enable nginx
```
Comment (lmilbaum) [inline nginx.sh:24]: Usually, the service is enabled first and started after.

**Review by lmilbaum — CHANGES_REQUESTED** [review]
I suggest you take a look at all the other PRs. You can find a lot of useful information.

File: bootstrap.sh (hunk 4 above)
Comment (MaromGold) [reply bootstrap.sh]:
```suggestion
#!/bin/bash
```

File: bootstrap.sh (hunk 1 above)
Comment (MaromGold) [reply bootstrap.sh]:
```suggestion
cd $HOME
```

File: bootstrap.sh (hunk 5 above)
Comment (MaromGold) [reply bootstrap.sh]:
```suggestion
```

File: ncdu.sh (hunk 3 above)
Comment (MaromGold) [reply ncdu.sh]:
```suggestion
fi

```

File: nginx.sh (hunk 6 above)
Comment (MaromGold) [reply nginx.sh:24]:
```suggestion
    sudo systemctl start nginx
```

Comment (MaromGold) [reply nginx.sh:24]: I was a bit conflicted about this part, didn't think it mattered. 
Apparently it does :) 
Thanks for the comment!

File: nginx.sh (hunk 2 above)
Comment (MaromGold) [reply nginx.sh]:
```suggestion
```

Comment (MaromGold) [reply nginx.sh]: I tried pushing changes from another station, and wanted to make sure that it worked. 
I forgot to delete this comment...
Thanks!

**Comment by MaromGold** [comment]
> The scripts look nice.
> All of them lack a short description of their purpose at the beginning.

In the last meeting, Liora specifically mentioned that this type of comment is redundant. 
She said: "WE don't need this type of commenting, we know how to read code". 
I actually had a brief comment, but I removed it after the last session.

**Comment by MaromGold** [comment]
> I suggest you take a look at all the other PRs. You can find a lot of useful information.

I'll do that. I didn't want to use other people's work before I finish "my own" assignment.

**Comment by MaromGold** [comment]
> The scripts look nice.
> All of them lack a short description of their purpose at the beginning.

Anyway, I added general comments in the beginning of each script... :)

File: bootstrap.sh [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+cd $HOME
+sudo wget -O /etc/yum.repos.d/jenkins.repo https://pkg.jenkins.io/redhat-stable/jenkins.repo
+sudo rpm --import https://pkg.jenkins.io/redhat-stable/jenkins.io.key
+yum install jenkins
+sudo systemctl start jenkins.service
+sudo systemctl enable jenkins.service
+
+tar -zxf git.tar.gz
+cd git-*
```
Comment (greycat001) [inline bootstrap.sh]: That is extremely unstable statement. '*' might easily yield multiple results resulting in too many arguments for the 'cd' command. If you want to get an unknown version of the package, try to filter out only the last result. This way, you will make sure that only one argument will pass to 'cd'.

Comment (MaromGold) [reply bootstrap.sh]:
```suggestion
cd  "$(\ls -1dt ./*/ | head -n 1)"
```

**Pull request merged**